tests/test_optim.py

```diff
@@ -496,22 +496,34 @@ def test_init_pars_sync_fixed_values_scipy(mocker):
 def test_init_pars_sync_fixed_values_minuit(mocker):
     opt = pyhf.optimize.minuit_optimizer()
 
-    # patch all we need
-    from pyhf.optimize import opt_minuit
-
-    minimizer = mocker.patch.object(opt_minuit, 'iminuit')
-    opt._get_minimizer(None, [9, 9, 9], [(0, 10)] * 3, fixed_vals=[(0, 1)])
-    assert minimizer.Minuit.from_array_func.call_args[1]['start'] == [1, 9, 9]
-    assert minimizer.Minuit.from_array_func.call_args[1]['fix'] == [True, False, False]
+    def f(x, y, z):
+        # Need a function of three parameters for below
+        return (x - 2) ** 2 + (y - 3) ** 2 + (z - 4) ** 2
+
+    # @kratsg, What is the need for mocker here?
+    # # patch all we need
+    # from pyhf.optimize import opt_minuit
+    #
+    # minimizer = mocker.patch.object(opt_minuit, 'iminuit')
+    minuit = opt._get_minimizer(f, [9, 9, 9], [(0, 10)] * 3, fixed_vals=[(0, 1)])
+    # assert minimizer.Minuit.from_array_func.call_args[1]['start'] == [1, 9, 9]
+    # assert minimizer.Minuit.from_array_func.call_args[1]['fix'] == [True, False, False]
+    assert list(minuit.values) == [1, 9, 9]
+    assert list(minuit.fixed) == [True, False, False]
```

Comment:
@kratsg Maybe a dumb question, but why did we need `mocker` here in the first place? Am I missing something that `mocker` does, or if I clean this up is using this `f` I've defined fine in your view?

Reply:
This was done so that we didn't need to do a proper function at all (e.g. if you mock out Minuit and just check what it gets called with -- without actually checking that it returns a proper object). The goal of this test was to test the behavior of our code, rather than Minuit's code.